src/serac/integrators/wrapper_integrator.hpp

```diff
@@ -221,4 +221,44 @@ class TransformedNonlinearFormIntegrator : public mfem::NonlinearFormIntegrator
   std::function<transformed_gradient_func> transformed_function_grad_;
 };
 
+/**
+ * @brief A class to hold a shared_ptr to an NonlinearformIntegrator that can be deleted without deleting the shared
+ * pointer improperly
+ */
+
+class PointerNonlinearFormIntegrator : public mfem::NonlinearFormIntegrator {
```

Comment:
It was originally used in the original newmark example to allow an `Integrator` to be shared by different `mfem::NonlinearForms` (ref)[https://github.com/LLNL/serac/pull/241#issuecomment-718139842]. When the forms go out-of-scope, it deletes all attached interrogators. This class is a light-weight wrapper that keeps `mfem::NonlinearForm`s happy while also properly ensuring that integrators can be shared (using `std::shared_ptr`) amongst `-Forms` or different solvers.

The original use of this structure in the newmark beta example allowed one to reuse the same integrator in combination with a `serac::TransformedNonlinearIntegrator` to do some variable substitutions depending on how you want to solve the 2nd order problem (solve for acceleration vs displacements). This could move some of the integrator-related logic that is currently coded in the rebuilding the `H_form` during updates.

Reply:
Gotcha. So we don't need it any more, but it's good to get in the repo for future use?

Reply:
Yes. It would be useful to keep for adjoint problems. If we don't use them, once we start doing adjoint problems we can throw it away or deprecate it.